packages/cli/src/commands/link/linkAll.js

```diff
@@ -4,39 +4,46 @@
 
 import {uniqBy} from 'lodash';
 import path from 'path';
-
+import {CLIError, logger} from '@react-native-community/cli-tools';
 import type {ConfigT} from 'types';
-
 import linkAssets from './linkAssets';
 import linkDependency from './linkDependency';
 
-import {CLIError} from '@react-native-community/cli-tools';
-
 const dedupeAssets = (assets: Array<string>): Array<string> =>
   uniqBy(assets, asset => path.basename(asset));
 
-async function linkAll(config: ConfigT) {
-  const projectAssets = config.assets;
-
-  const assets = dedupeAssets(
-    Object.keys(config.dependencies).reduce(
-      (acc, dependency) => acc.concat(config.dependencies[dependency].assets),
-      projectAssets,
-    ),
-  );
+type Options = {
+  linkDeps?: boolean,
+  linkAssets?: boolean,
+};
 
+async function linkAll(config: ConfigT, options: Options) {
   try {
-    Object.keys(config.dependencies).forEach(async key => {
-      const dependency = config.dependencies[key];
-      if (dependency.hooks.prelink) {
-        await dependency.hooks.prelink();
-      }
-      await linkDependency(config.platforms, config.project, dependency);
-      if (dependency.hooks.postlink) {
-        await dependency.hooks.postlink();
-      }
-    });
-    await linkAssets(config.platforms, config.project, assets);
+    if (options.linkDeps) {
+      logger.debug('Linking all dependencies');
+      Object.keys(config.dependencies).forEach(async key => {
```

Comment:
I think we should print some kind of deprecation warning here.

Reply:
Maybe not deprecation, but a warning message about an alternative would be nice way to let people know about autolinking 👍